### PR TITLE
feat: add choices to switch configurator

### PR DESCRIPTION
**Pull request by jholt1**
Add ability to change component in the configurator

### CODE REVIEW

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 1]
```diff
@@ -1,54 +1,77 @@
...
-import '/node_modules/@astrojs/starlight/style/shiki.css';
-import '/node_modules/@astrojs/starlight/style/util.css';
-import '../../../../../styles/custom.css';
-import ExampleConfigure from '../../../../../components/ExampleConfigure.astro';
+import "/node_modules/@astrojs/starlight/style/props.css";
+import "/node_modules/@astrojs/starlight/style/reset.css";
+import "/node_modules/@astrojs/starlight/style/shiki.css";
+import "/node_modules/@astrojs/starlight/style/util.css";
+import "../../../../../styles/custom.css";
+import ExampleConfigure from "../../../../../components/ExampleConfigure.astro";
```
Comment (andij) [inline src/pages/configure/components/[component]/[type]/index.astro:7]: Prettier has altered these and other lines in this file.

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 2]
```diff
@@ -1,54 +1,77 @@
 ---
-import '/node_modules/@astrojs/starlight/style/props.css';
-import '/node_modules/@astrojs/starlight/style/reset.css';
-import '/node_modules/@astrojs/starlight/style/shiki.css';
-import '/node_modules/@astrojs/starlight/style/util.css';
-import '../../../../../styles/custom.css';
-import ExampleConfigure from '../../../../../components/ExampleConfigure.astro';
+import "/node_modules/@astrojs/starlight/style/props.css";
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:2]:
```suggestion
import '/node_modules/@astrojs/starlight/style/props.css';
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 3]
```diff
@@ -1,54 +1,77 @@
...
+import "/node_modules/@astrojs/starlight/style/reset.css";
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:3]:
```suggestion
import '/node_modules/@astrojs/starlight/style/reset.css';
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 4]
```diff
@@ -1,54 +1,77 @@
...
+import "/node_modules/@astrojs/starlight/style/shiki.css";
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:4]:
```suggestion
import '/node_modules/@astrojs/starlight/style/shiki.css';
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 5]
```diff
@@ -1,54 +1,77 @@
...
+import "/node_modules/@astrojs/starlight/style/util.css";
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:5]:
```suggestion
import '/node_modules/@astrojs/starlight/style/util.css';
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 6]
```diff
@@ -1,54 +1,77 @@
...
+import "../../../../../styles/custom.css";
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:6]:
```suggestion
import '../../../../../styles/custom.css';
```

File: src/pages/configure/components/[component]/[type]/index.astro (hunk 1 above)
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:7]:
```suggestion
import ExampleConfigure from '../../../../../components/ExampleConfigure.astro';
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 7]
```diff
@@ -1,54 +1,77 @@
...
+          },
         };
       });
     } else {
       return {
         params: {
           component: tag.name,
-          type: 'standard'
-        }
+          type: "standard",
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:62]:
```suggestion
          type: 'standard',
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 8]
```diff
@@ -1,54 +1,77 @@
...
 const { component, type } = Astro.params;
+const { allTagNames } = Astro.props;
 const exampleFileMatcher = `/assets/components/${component}/${type}.html`;
 
-const examples = import.meta.glob("../../../../../assets/components/**/*.html", { import: 'default', as: 'raw', eager: true});
-const htmlFile = Object.keys(examples).find((example) => example?.endsWith(exampleFileMatcher));
+const examples = import.meta.glob(
+  "../../../../../assets/components/**/*.html",
+  { import: "default", as: "raw", eager: true },
+);
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:17]:
```suggestion
const examples = import.meta.glob("../../../../../assets/components/**/*.html", { import: "default", as: "raw", eager: true });
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 9]
```diff
@@ -1,54 +1,77 @@
...
+const htmlFile = Object.keys(examples).find((example) =>
+  example?.endsWith(exampleFileMatcher),
+);
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:20]:
```suggestion
const htmlFile = Object.keys(examples).find((example) => example?.endsWith(exampleFileMatcher));
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 10]
```diff
@@ -1,54 +1,77 @@
...
 const htmlContent = await examples[htmlFile];
 
 export function getStaticPaths() {
-  const examples = import.meta.glob("../../../../../assets/components/**/*.html");
+  const examples = import.meta.glob(
+    "../../../../../assets/components/**/*.html",
+  );
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:26]:
```suggestion
  const examples = import.meta.glob("../../../../../assets/components/**/*.html");
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 11]
```diff
@@ -1,54 +1,77 @@
...
   const getComponentExample = (fileMatcher) => {
-    return Object.keys(examples)?.filter((example) => example.indexOf(fileMatcher) > -1);
-  } 
+    return Object.keys(examples)?.filter(
+      (example) => example.indexOf(fileMatcher) > -1,
+    );
+  };
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:31]:
```suggestion
    return Object.keys(examples)?.filter((example) => example.indexOf(fileMatcher) > -1)};
```

File: src/pages/configure/components/[component]/[type]/index.astro [hunk 12]
```diff
@@ -1,54 +1,77 @@
...
     const componentName = tag.name;
     const exampleComponentMatcher = `/assets/components/${componentName}/`;
     const componentExamples = getComponentExample(exampleComponentMatcher);
-    if(componentExamples) {
+    if (componentExamples) {
       return componentExamples.map((componentExample) => {
-        const type = componentExample?.split(exampleComponentMatcher)[1].replaceAll('.html', '');
+        const type = componentExample
+          ?.split(exampleComponentMatcher)[1]
+          .replaceAll(".html", "");
```
Comment (PtitBen) [inline src/pages/configure/components/[component]/[type]/index.astro:47]:
```suggestion
        const type = componentExample?.split(exampleComponentMatcher)[1].replaceAll(".html", "");
```

**Review by PtitBen — COMMENTED** [review]
I suggested the changes Drew mentioned ;)

**Review by PtitBen — APPROVED** [review]

**Pull request merged**